rust/src/connection/network/transmitter/transaction.rs

```diff
@@ -241,60 +242,46 @@ impl TransactionTransmitter {
             on_close: Default::default(),
             callback_handler_sink,
         };
-        tokio::spawn(Self::dispatch_loop(
-            queue_source,
-            request_sink,
-            collector.clone(),
-            on_close_callback_source,
-            shutdown_signal,
-        ));
+        let cloned_collector = collector.clone();
+        tokio::task::spawn_blocking(move || {
+            Self::dispatch_loop(queue_source, request_sink, cloned_collector, on_close_callback_source, shutdown_signal)
+        });
```

Comment:
Minor, but this is the common way of moving a clone into a closure:
```suggestion
        tokio::task::spawn_blocking({
            let collector = collector.clone();
            move || Self::dispatch_loop(queue_source, request_sink, collector, on_close_callback_source, shutdown_signal)
        });
```